packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { _formatString } from '@internal/acs-ui-common';
+import { RemoteParticipantState } from '@internal/calling-stateful-client';
+import { useEffect, useState } from 'react';
+import { useLocale } from '../../localization';
+import { useSelector } from '../hooks/useSelector';
+import { getRemoteParticipants } from '../selectors/baseSelectors';
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipantsObject = useSelector(getRemoteParticipants);
+  const remoteParticipants = Object.values(remoteParticipantsObject ?? {}).filter((p) => {
+    /**
+     * We need to make sure remote participants are in the connected state. If they are not
+     * they might not have their displayName set in the call state just yet.
+     */
+    if (p.state === 'Connected') {
+      return p;
+    }
+    return;
+  });
+  const [currentParticipants, setCurrentParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+    if (remoteParticipants.length < currentParticipants.length) {
+      //someone left
+      const whoLeft = currentParticipants.filter((p) => {
+        if (!remoteParticipants.includes(p)) {
+          return p;
+        }
+        return;
+      });
+      setParticipantEventString(
+        createAnnouncmentString(
+          locale.participantLeftNoticeString,
+          locale.unnamedParticipantChangedString,
+          locale.participantsJoinedOverflowString,
+          whoLeft
+        )
+      );
+      setCurrentParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > currentParticipants.length) {
+      // someone joined
```

Comment:
The current code doesn't handle people joining and leaving at the same time (we don't have any place where we create a string that handles both cases together).